cupy/random/_generator_api.pyx

```diff
@@ -351,6 +360,98 @@ class Generator:
 
         return y
 
+    def gamma(self, shape, scale=1.0, size=None):
+        """Returns an array of samples drawn from a gamma distribution.
+
+        Args:
+            shape (float or array_like of float): The shape of the
+                gamma distribution.  Must be non-negative.
+            scale (float or array_like of float): The scale of the
+                gamma distribution.  Must be non-negative.
+                Default equals to 1
+            size (int or tuple of ints): The shape of the array.
+                If ``None``, a zero-dimensional array is generated.
+
+        .. seealso::
+            - :func:`cupy.random.gamma` for full documentation
+            - :meth:`numpy.random.Generator.gamma`
+        """
+        return self.standard_gamma(shape, size) * scale
```

Comment:
```suggestion
        if size is None:
            size = cupy.broadcast(shape, scale).shape
        y = self.standard_gamma(shape, size)
        y *= scale
        return y
```

- `y *= scale` detects the bug below and might improve the performance. 
- `size` has to be inferred here if `scale` is an array.

```
for xp in (numpy, cupy):
    print(xp.__name__, xp.random.default_rng(0).gamma(2, scale=xp.ones(5)))
```
results
```
numpy [1.83431007 2.63769797 1.06635807 3.98054168 0.91323087]
cupy [0.77175678 0.77175678 0.77175678 0.77175678 0.77175678]
```